### ISSUE TITLE
YAML generated file encoding issue with Ruby 1.9.3

### ISSUE THREAD

**Issue by pmorillon** [grid5000/g5k-checks#3]
When generating yaml file for the reference API : 

```
$ g5k-checks -m api
```

There is an encoding issue : 

``` YAML
block_devices:
  sda:
    device: sda
    size: 600127266816
    !binary "bW9kZWw=": !binary |-
      U1Q2MDBNTTAwMDY=
    !binary "cmV2": !binary |-
      TFMwQQ==
  sdb:
    device: sdb
    size: 600127266816
    !binary "bW9kZWw=": !binary |-
      U1Q2MDBNTTAwMDY=
    !binary "cmV2": !binary |-
      TFMwQQ==
```

With Ruby 1.9.3, the default YAML engine is Psych, we can fix this by using the old Engine 'Sick' : 

``` Ruby
 YAML::ENGINE.yamler='syck'
```

But this engine is removed for Ruby >= 2.0.0...

**Comment by priteau** [comment]
Hello @pmorillon,

I had the same problem when I first started using g5k-checks in Chameleon using an Ubuntu environment, and I developed a fix at that time. Later we moved to CentOS 7 which doesn't show this issue, so I didn't push the patch as I am not sure it is the best solution… but it might be helpful to you.

https://github.com/ChameleonCloud/g5k-checks/commit/a46022a9b913302af3ca5862e1aad8ac47647297

**Comment by pmorillon** [comment]
@priteau patch seems to be better, we have to test with our Debian environments.

**Comment by pmorillon** [comment]
`scripts/getconfig.rb` finally use the JSON format.

**Comment by priteau** [comment]
@pmorillon What solution did you end up using for fixing this bug?

**Comment by pmorillon** [comment]
I used the JSON output.

**Comment by SimonDelamare** [comment]
I've cherry-picked priteau's patch, it is available in g5k-check 0.4.15.

(BTW, Pierre, this version also include a fix for the processor / version entry, you may have to update your nodes description if you use it)
